workers/unity/Assets/Fps/Scripts/UI/Hud/Hud.cs

```diff
@@ -52,8 +52,8 @@ private void OnEnable()
         private void OnRespawn(Empty obj)
         {
             // Hide respawn screen
-            inGameUIController.RespawnScreen.SetActive(false);
-            inGameUIController.Hud.SetActive(true);
+            inGameScreenManager.RespawnScreen.SetActive(false);
```

Comment:
What happens if `inGameScreenManager` is null?

Reply:
then your game won't work

the question is: if the screens are not set correctly, the whole game won't work correctly. 
I would prefer to just propagate the null pointer exception up instead of null checking everything and then throwing a new error